pinot-controller/src/main/java/org/apache/pinot/controller/api/debug/TableDebugInfo.java

```diff
@@ -0,0 +1,212 @@
+  @JsonProperty("segmentDebugInfos")
+  private final List<SegmentDebugInfo> _segmentDebugInfos;
+
+  @JsonProperty("serverDebugInfos")
+  private final List<ServerDebugInfo> _serverDebugInfos;
+
+  @JsonProperty("brokerDebugInfos")
+  private final List<BrokerDebugInfo> _brokerDebugInfos;
+
+  @JsonCreator
+  public TableDebugInfo(String tableName, TableSizeSummary tableSizeSummary, List<SegmentDebugInfo> segmentDebugInfos,
+      List<ServerDebugInfo> serverDebugInfos, List<BrokerDebugInfo> brokerDebugInfos, int numBrokers, int numServers,
+      int numSegments) {
+    _tableName = tableName;
+    _tableSizeSummary = tableSizeSummary;
+    _segmentDebugInfos = segmentDebugInfos;
+    _serverDebugInfos = serverDebugInfos;
+    _brokerDebugInfos = brokerDebugInfos;
+
+    _numBrokers = numBrokers;
+    _numServers = numServers;
+    _numSegments = numSegments;
+  }
+
+  public String getTableName() {
+    return _tableName;
+  }
+
+  public TableSizeSummary getTableSize() {
+    return _tableSizeSummary;
+  }
+
+  public int getNumSegments() {
+    return _numSegments;
+  }
+
+  public int getNumServers() {
+    return _numServers;
+  }
+
+  public int getNumBrokers() {
+    return _numBrokers;
+  }
+
+  public List<SegmentDebugInfo> getSegmentDebugInfos() {
+    return _segmentDebugInfos;
+  }
+
+  public List<ServerDebugInfo> getServerDebugInfos() {
+    return _serverDebugInfos;
+  }
+
+  public List<BrokerDebugInfo> getBrokerDebugInfos() {
+    return _brokerDebugInfos;
+  }
+
+  public static class SegmentDebugInfo {
+    private final String _segmentName;
+    private final List<IsEvState> _iSEvStates;
+
+    public SegmentDebugInfo(String name, List<IsEvState> iSEvStates) {
+      _segmentName = name;
+      _iSEvStates = iSEvStates;
+    }
+
+    public String getSegmentName() {
+      return _segmentName;
+    }
+
+    public List<IsEvState> getSegmentStateInServer() {
+      return _iSEvStates;
+    }
+  }
+
+  public static class IsEvState {
+    private final String _serverName;
+    private final String _idealStateStatus;
+    private final String _externalViewStatus;
+
+    public IsEvState(String name, String idealStateStatus, String externalViewStatus) {
+      _serverName = name;
+      _idealStateStatus = idealStateStatus;
+      _externalViewStatus = externalViewStatus;
+    }
+
+    public String getServerName() {
+      return _serverName;
+    }
+
+    public String getIdealStateStatus() {
+      return _idealStateStatus;
+    }
+
+    public String getExternalViewStatus() {
+      return _externalViewStatus;
+    }
+  }
+
+  public static class ServerDebugInfo {
+    private final int _numErrors;
+    private final int _numNewMessages;
+    private final int _numReadMessages;
+    private final String _serverName;
+
+    public ServerDebugInfo(String serverName, int numErrors, int numNewMessages, int numReadMessages) {
+      _serverName = serverName;
+      _numErrors = numErrors;
+      _numNewMessages = numNewMessages;
```

Comment:
What's the difference between new and read messages? Some documentations would be good